src/main/java/org/springframework/data/domain/ManagedTypes.java

```diff
@@ -77,6 +79,31 @@ static ManagedTypes fromIterable(Iterable<? extends Class<?>> types) {
 		return types::forEach;
 	}
 
+	/**
+	 * Factory method used to construct {@link ManagedTypes} from the given, required {@link Iterable} of {@link String
+	 * type names}.
+	 *
+	 * @param types {@link Iterable} of {@literal class names} used to initialize the {@link ManagedTypes}; must not be
+	 *          {@literal null}.
+	 * @return new instance of {@link ManagedTypes} initialized the given, required {@link Iterable} of {@link Class
+	 *         types}.
+	 * @throws IllegalStateException if class cannot be loaded.
+	 * @see java.lang.Iterable
+	 * @see #fromStream(Stream)
+	 * @see #fromSupplier(Supplier)
+	 */
+	static ManagedTypes fromClassNames(Iterable<String> types) {
```

Comment:
Likely we should use the bean class loader instead of the one from `ManagedTypes`.